pkg/liqo-controller-manager/resource-request-controller/resource-monitors/external-monitor.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourcemonitors
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog/v2"
+)
+
+// ExternalResourceMonitor is an object that keeps track of the cluster's resources.
+type ExternalResourceMonitor struct {
+	ResourceReaderClient
+}
+
+// NewExternalMonitor creates a new ExternalResourceMonitor.
+func NewExternalMonitor(address string) (*ExternalResourceMonitor, error) {
+	conn, err := grpc.Dial(address, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		klog.Errorf("Could not connect to external resource monitor at %s: %s", address, err)
+		return nil, err
+	}
+	client := NewResourceReaderClient(conn)
+	return &ExternalResourceMonitor{
+		ResourceReaderClient: client,
+	}, nil
+}
+
+// Register sets an update notifier.
+func (m *ExternalResourceMonitor) Register(ctx context.Context, notifier ResourceUpdateNotifier) {
+	stream, err := m.ResourceReaderClient.Subscribe(ctx, &SubscribeRequest{})
+	if err != nil {
+		klog.Errorf("grpc error while subscribing: %s", err)
+	}
+	go func() {
+		for {
+			_, err := stream.Recv()
+			if errors.Is(err, io.EOF) {
+				break
```

Comment:
Can we log something when the function exists? It can be at higher verbosity, but it may be useful for debugging.

And, second, what will happen if the remote monitor pod restarts? The controller manager will reconnect to it? Will it restart? It is not clear to me.

Reply:
It reconnects with exponential backoff per gRPC docs: https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md, https://stackoverflow.com/a/39357099